addons/stock_account/wizard/stock_valuation_layer_revaluation.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools import float_is_zero
+
+
+class StockValuationLayerRevaluation(models.TransientModel):
+    _name = 'stock.valuation.layer.revaluation'
+    _description = "Wizard model to reavaluate a stock inventory for a product"
+
+    @api.model
+    def default_get(self, default_fields):
+        res = super().default_get(default_fields)
+        if res.get('product_id'):
+            product = self.env['product.product'].browse(res['product_id'])
+            accounts = product.product_tmpl_id.get_product_accounts()
+            if product.categ_id.property_cost_method == 'standard':
+                raise UserError(_("You can't revaluate a product with a standard cost method"))
+            if product.quantity_svl <= 0:
+                raise UserError(_("You cannot revalue an empty or negative stock"))
+            if 'account_journal_id' not in res and 'account_journal_id' in default_fields:
+                res['account_journal_id'] = accounts['stock_journal'].id
+        return res
+
+    company_id = fields.Many2one('res.company', "Company", readonly=True, required=True)
+    currency_id = fields.Many2one('res.currency', "Currency", related='company_id.currency_id', required=True)
+
+    product_id = fields.Many2one('product.product', "Related product", required=True, check_company=True)
```

Comment:
you never calls "self._check_company()" so the `check_company` attributes only set domain
i would set "_check_company_auto = True" on both models